# Thorough EDA of the WiDs Climate Challenge Datasets

The purpose of this dataset is to conduct a throrough explaratory data analysis of the train and test datasests as well as target of this competition.

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn import preprocessing, model_selection

from sklearn.ensemble import RandomForestRegressor
from sklearn import svm as svm
import lightgbm as lgb
from sklearn.ensemble import GradientBoostingRegressor as grb

# Reading data
___

In [ ]:
# Reading the clean train dataset
train_df = pd.read_csv("data/train_data.csv")
print(f'The shape of the dataset is: {train_df.shape}')
train_df.head()

The shape of the dataset is: (375734, 246)


index  lat       lon startdate  contest-pevpr-sfc-gauss-14d__pevpr  \
0      0  0.0  0.833333    9/1/14                              237.00   
1      1  0.0  0.833333    9/2/14                              228.90   
2      2  0.0  0.833333    9/3/14                              220.69   
3      3  0.0  0.833333    9/4/14                              225.28   
4      4  0.0  0.833333    9/5/14                              237.24   

   nmme0-tmp2m-34w__cancm30  nmme0-tmp2m-34w__cancm40  \
0                     29.02                     31.64   
1                     29.02                     31.64   
2                     29.02                     31.64   
3                     29.02                     31.64   
4                     29.02                     31.64   

   nmme0-tmp2m-34w__ccsm30  nmme0-tmp2m-34w__ccsm40  nmme0-tmp2m-34w__cfsv20  \
0                    29.57                    30.73                    29.71   
1                    29.57                    30.73                    29.71   
2                    29.57                    30.73                    29.71   
3                    29.57                    30.73                    29.71   
4                    29.57                    30.73                    29.71   

   ...  wind-vwnd-925-2010-11  wind-vwnd-925-2010-12  wind-vwnd-925-2010-13  \
0  ...                 -27.68                 -37.21                   8.32   
1  ...                 -21.13                 -36.57                   8.77   
2  ...                 -10.72                 -34.16                   6.99   
3  ...                   0.33                 -31.04                   6.17   
4  ...                   9.83                 -31.80                   7.47   

   wind-vwnd-925-2010-14  wind-vwnd-925-2010-15  wind-vwnd-925-2010-16  \
0                   9.56                  -2.03                  48.13   
1                  21.17                   4.44                  48.60   
2                  32.16                   5.01                  48.53   
3                  39.66                  -1.41                  50.59   
4                  38.62                  -5.21                  54.73   

   wind-vwnd-925-2010-17  wind-vwnd-925-2010-18  wind-vwnd-925-2010-19  \
0                  28.09                 -13.50                  11.90   
1                  27.41                 -23.77                  15.44   
2                  19.21                 -33.16                  15.11   
3                   8.29                 -37.22                  18.24   
4                  -2.58                 -42.30                  21.91   

   wind-vwnd-925-2010-20  
0                   4.58  
1                   3.42  
2                   4.82  
3                   9.74  
4                  10.95  

[5 rows x 246 columns]

In [ ]:
# Reading the clean test dataset
test_df = pd.read_csv("data/test_data.csv")
print(f'The shape of the dataset is: {test_df.shape}')
test_df.head()

The shape of the dataset is: (31354, 245)


index  lat       lon startdate  contest-pevpr-sfc-gauss-14d__pevpr  \
0  375734  0.0  0.833333   11/1/22                              339.88   
1  375735  0.0  0.833333   11/2/22                              334.63   
2  375736  0.0  0.833333   11/3/22                              337.83   
3  375737  0.0  0.833333   11/4/22                              345.81   
4  375738  0.0  0.833333   11/5/22                              357.39   

   nmme0-tmp2m-34w__cancm30  nmme0-tmp2m-34w__cancm40  \
0                     30.88                     30.92   
1                     30.88                     30.92   
2                     30.88                     30.92   
3                     30.88                     30.92   
4                     30.88                     30.92   

   nmme0-tmp2m-34w__ccsm30  nmme0-tmp2m-34w__ccsm40  nmme0-tmp2m-34w__cfsv20  \
0                    29.17                    31.02                    29.47   
1                    29.17                    31.02                    29.47   
2                    29.17                    31.02                    29.47   
3                    29.17                    31.02                    29.47   
4                    29.17                    31.02                    29.47   

   ...  wind-vwnd-925-2010-11  wind-vwnd-925-2010-12  wind-vwnd-925-2010-13  \
0  ...                 -19.28                 -39.77                 -29.25   
1  ...                 -19.58                 -43.14                 -28.62   
2  ...                 -13.73                 -44.22                 -27.67   
3  ...                  -7.97                 -49.47                 -19.32   
4  ...                  -0.80                 -56.07                  -9.89   

   wind-vwnd-925-2010-14  wind-vwnd-925-2010-15  wind-vwnd-925-2010-16  \
0                  40.88                  -8.31                  14.91   
1                  45.37                  -5.42                  16.97   
2                  49.76                  -1.31                  21.44   
3                  52.62                  -0.44                  21.65   
4                  51.23                  -7.57                  19.86   

   wind-vwnd-925-2010-17  wind-vwnd-925-2010-18  wind-vwnd-925-2010-19  \
0                 -24.62                  31.05                 -23.69   
1                 -23.94                  28.84                 -20.61   
2                 -19.06                  26.85                 -16.78   
3                 -23.12                  23.70                 -18.62   
4                 -30.56                  20.66                 -25.08   

   wind-vwnd-925-2010-20  
0                   6.27  
1                  14.16  
2                  13.42  
3                  10.69  
4                  19.64  

[5 rows x 245 columns]

In [ ]:
# Readig the sample submission file
sample_sub = pd.read_csv('data/sample_solution.csv')
sample_sub.head()

contest-tmp2m-14d__tmp2m   index
0                 27.073876  375734
1                 25.109308  375735
2                 22.557390  375736
3                 25.572875  375737
4                 20.781073  375738

# Comprehensive Meaning of Features in the datasets
___

In [ ]:
# Get the list of all column names from headers
count = 0
for column_headers in train_df.columns:
    print(f'{column_headers}')

In [ ]:
# locs = train_df.loc[train_df['startdate'] == '9/1/14']
# locs1 = locs[['lat', 'lon', 'climateregions__climateregion']]
# locs1.to_csv('locations.csv')

# nmme = train_df[[
#     'startdate',
#     'nmme0-tmp2m-34w__nmme0mean',
#     'nmme-tmp2m-56w__nmmemean',
#     'nmme-prate-34w__nmmemean',
#    'nmme0-prate-56w__nmme0mean',
#    'nmme0-prate-34w__nmme0mean',
#    'nmme-prate-56w__nmmemean',
#     'nmme-tmp2m-34w__nmmemean',
#     'nmme0mean',
# ]]
# nmme.to_csv('nmme_means.csv')

## **The 245 features**

### **General Features**
---

- Ft 1 : **index**
    - Each row in the data corresponds to a single location and a single start date for the two-week period.
    - The data provided for each date is a combination of:
        - weather and climate information for a number of US locations,
        - for a number of start dates for the two-week observation,
        - as well as the forecasted temperature and precipitation from a number of weather forecast models
    - The train dataset has a shape of (375734, 246) while the test dataset is much smaller at (31354, 245).
    
- Ft 2 & Ft 3 & Ft 100 : **lat & lon & climateregions__climateregion**
    - The Latitude and Longitude features
    - We have 514 locations in total and each location has data for 731 startdates ranging from 9th Jan 2014 to 31st Aug 2016.
    - A climate region has several locations ... view locations [here](https://drive.google.com/file/d/1SabOmKMFM0keZ1k7y5xbeRAcwcRZCA34/view?usp=share_link)
    
    > - What interesting features could we derive from these three?
    > - Can we combine these features to make it more meaningful?
     
- Ft 4 : **startdate**
    - The startdate of a two-week observation period
    - There are 731 startdates ranging from 1st Sep 2014 to 31st Aug 2016.
    
    > - What interesting features could we derive from the date ranges with regard to long term forecasting?
    > - Maybe months, quaters, seasons?

### **Model Forecast Features**
---

- "NMME" stands for "North American Multi-Model Ensemble," which is an enesemble model for multiple climate models.
- The following prefixes i.e. cancm30, cancm40, ccsm30, ccsm40, cfsv20, gfdlflora0, gfdlflorb0, gfdl0, nasa0 refer to the differnet ensemble members.
- The nmme0mean feature is the average forecast across these models.
- It seems that for these features, the values remain constant over certain date ranges and then vary slightly over the next set of ranges. To mean they change seasonaly
- View an extraction of these features against the startdate column [here](https://drive.google.com/file/d/1TtndCkgl58CHkSo-1vjD1mFXxNzE_T42/view?usp=sharing)

> Since we have the ensemble mean for the different forecasts, do we drop the other features used to get the mean and just work with mean?
> Do the tmp2m model forecast features speak to high feature importance with regard to the target? 
    
#### **Tmp2m model forcasts**

- Ft 6 to Ft 15: **nmme0-tmp2m-34w__ ...**
    - Weeks 3-4 weighted average of most recent monthly NMME model forecasts for tmp2m
- Ft 74 to Ft 83: **nmme-tmp2m-34w__ ...**
    - Weeks 3-4 weighted average of monthly NMME model forecasts for target label, contest-tmp2m-14d__tmp2m
- Ft 17 to Ft 26: **nmme-tmp2m-56w__ ...**
    - Weeks 5-6 weighted average of monthly NMME model forecasts for target label, contest-tmp2m-14d__tmp2m

#### **Precipitation model forcasts**

- Ft 28 to Ft 37: **nmme-prate-34w__ ...**
    - Weeks 3-4 weighted average of monthly NMME model forecasts for precipitation
- Ft 49 to Ft 58: **nmme0-prate-34w__ ...**
    - Weeks 3-4 weighted average of most recent monthly NMME model forecasts for precipitation
- Ft 62 to Ft 71: **nmme-prate-56w__ ...**
    - Weeks 5-6 weighted average of monthly NMME model forecasts for precipitation
- Ft 39 to Ft 48: **nmme0-prate-56w__ ...**
    - Weeks 5-6 weighted average of most recent monthly NMME model forecasts for precipitation
    
#### **Average model forcasts**

- Ft 90 to Ft 99: **cancm30, cancm40, ccsm30*, ccsm40, cfsv20, gfdlflora0, gfdlflorb0, fdl0, nasa0, nmme0mean**
    - The most recent forecasts from different weather models
    - nmme0mean is their average
    
### **Evaporation Features**
---
- Evaporation refers to the conversion of water from its liquid state into water vapor (gas) through the process of heating and vaporization.
- This occurs when the surface temperature of water reaches the boiling point and the water vapor rises into the atmosphere. 
- Evaporation can affect temperature in several ways:
    - Cooling effect: When water evaporates, it removes heat from the surrounding air, causing the air temperature to decrease. This is why sweating cools down the body.
    - Latent heat of vaporization: The conversion of water from its liquid state into water vapor releases latent heat, which is a form of energy. This release of latent heat can lead to a cooling effect on the air, which can lower the temperature.
    - Atmospheric moisture: The evaporation of water from the surface increases the atmospheric moisture, which can affect temperature by trapping heat in the atmosphere through the greenhouse effect.
    
- Ft 5 : **contest-pevpr-sfc-gauss-14d__pevpr**
    - Potential evaporation
    
### **Wind Features**
___

- Ft 16 : **contest-wind-h10-14d__wind-hgt-10**
- Ft 207 to Ft 216: **wind-hgt-10-2010-1 to wind-hgt-10-2010-10**
    - Geopotential height at 10 millibars
- Ft 38 : **contest-wind-h100-14d__wind-hgt-100**
- Ft 217 to Ft 226: **wind-hgt-100-2010-1 to wind-hgt-100-2010-10**
    - Geopotential height at 100 millibars
- Ft 89 : **contest-wind-h500-14d__wind-hgt-500**
    - Geopotential height at 500 millibars
- Ft 167 to Ft 176: **wind-hgt-500-2010-1 to wind-hgt-500-2010-10**
- Ft 87 : **contest-wind-h850-14d__wind-hgt-850**
- Ft 147 to fT 156: **wind-hgt-850-2010-1 to wind-hgt-850-2010-10**
    - Geopotential height at 850 millibars

<br>

- Ft 61 : **contest-wind-vwnd-925-14d__wind-vwnd-925**
- Ft 227 to Ft 246: **wind-vwnd-925-2010-1 to wind-vwnd-925-2010-20**
    - Longitudinal wind at 925 millibars
- Ft 85 : **contest-wind-vwnd-250-14d__wind-vwnd-250**
- Ft 102 to Ft 121: **wind-vwnd-250-2010-1 to wind-vwnd-250-2010-20**
    - Longitudinal wind at 250 millibars

<br>

- Ft 73 : **contest-wind-uwnd-250-14d__wind-uwnd-250**
- Ft 122 to Ft 141: **wind-uwnd-250-2010-1 to wind-uwnd-250-2010-20**
    - Zonal wind at 250 millibars
- Ft 88 : **contest-wind-uwnd-925-14d__wind-uwnd-925**
- Ft 187 to Ft 206 : **wind-uwnd-925-2010-1 to wind-uwnd-925-2010-20**
    - Zonal wind at 925 millibars
    
## **Humidity Features**
___
- Humidity refers to the amount of water vapor present in the air. 
- It is a measure of the amount of moisture in the atmosphere and is expressed as a percentage of the maximum amount of moisture that the air can hold at a given temperature.
- Humidity can affect atmospheric temperature in several ways:
    - Latent Heat: When air becomes saturated with water vapor, it releases latent heat as the water vapor condenses into droplets, such as clouds or fog. This release of latent heat can lead to a cooling effect on the air, which can lower the atmospheric temperature.
    - Evaporation: When water evaporates, it removes heat from the surrounding air, causing the atmospheric temperature to decrease.
    - Heat Index: The heat index, also known as the "feels like" temperature, takes into account both the atmospheric temperature and humidity to provide a more accurate representation of how hot it feels outside. When the humidity is high, the body's ability to cool down through sweating is reduced, making it feel warmer than the actual atmospheric temperature.
    
<br> 

- Ft 27 : **contest-rhum-sig995-14d__rhum**
    - Relative humidity
    
## **Sea Level Pressure Features**
___
- Sea level pressure, also known as atmospheric pressure at sea level, is the atmospheric pressure at a reference height of mean sea level.
- It can affect temperature in a few ways:
    - Adiabatic Lapse Rate: Air temperature decreases with height in the lower atmosphere according to the adiabatic lapse rate. Changes in sea level pressure can cause changes in the atmospheric pressure at a given height, which can in turn cause changes in temperature.
    - Pressure-Driven Circulation: Sea level pressure is a key factor in driving large-scale atmospheric circulation patterns, such as the Hadley cell and the polar cell. These patterns can affect temperature by transporting warm air from equatorial regions to the poles, and cold air from the poles to the equator.
    - Latitude-Dependent Pressure Patterns: Sea level pressure is also affected by latitude, with lower pressure near the equator and higher pressure near the poles. This latitudinal pressure gradient can drive temperature changes by affecting the flow of air masses between different regions.
- Overall, sea level pressure is an important factor that can impact temperature by affecting atmospheric circulation patterns, atmospheric pressure gradients, and the adiabatic lapse rate.

> Find relationship between this feature, the latitude feature and the target

<br>

- Ft 60 : **contest-slp-14d__slp**
    - File containing sea level pressure (slp) 
    
## **Pressure Features**
___
- Ways that air pressure can affect temperature:
    - Adiabatic Lapse Rate: In the atmosphere, temperature decreases with height due to the adiabatic lapse rate. This is a result of the expansion of air as it rises and the decrease in pressure that occurs with increasing altitude. Conversely, air that is compressed by a decrease in pressure will experience an increase in temperature.
    - Gas Law Relationships: The relationship between air pressure and temperature is described by the gas laws, such as Boyle's Law and Charles's Law. These laws describe how changes in pressure and temperature are related for a given volume of gas. For example, if the pressure of a gas is increased, its temperature will also increase.

<br>

- Ft 72 : **contest-pres-sfc-gauss-14d__pres**
    - Pressure
    
## **Precipitable Water & Precipitation Features**
___
- Precipitation is any form of water, such as rain, snow, sleet, or hail, that falls from the atmosphere to the surface of the Earth.
- Precipitable water is a measure of the amount of water vapor present in a vertical column of the atmosphere, from the surface to the top of the atmosphere. It is expressed in units of millimeters (mm) or centimeters (cm) and represents the amount of water that would fall to the ground as precipitation if all the water vapor in the column were to condense.
- Precipitation occurs when water vapor in the atmosphere condenses into clouds, and then falls to the ground due to the force of gravity. 
- Precipitation can have both direct and indirect effects on temperature:
    - Direct effect: Precipitation itself can cause a change in temperature. For example, when rain falls, it can bring cooler air down with it, causing a drop in temperature. On the other hand, snow can reflect sunlight, which can help to keep temperatures cooler, especially in the winter.
    - Indirect effect: Precipitation can also affect temperature by altering the amount of solar radiation that reaches the surface of the Earth. For example, if there is a lot of cloud cover associated with a precipitation event, this can reduce the amount of solar radiation that reaches the ground, leading to cooler temperatures. Similarly, if there is less cloud cover, more solar radiation can reach the surface, causing temperatures to warm.
- Overall, the effect of precipitation on temperature can be complex and depends on many factors, including the type of precipitation, the time of year, and the specific climate of a given region. 

<br>

- Ft 84 : **contest-prwtr-eatm-14d__prwtr**
    - Precipitable water for entire atmosphere
- Ft 86 : **contest-precip-14d__precip**
    - Measured precipitation

    
## **Elevation Features**
___
- Elevation refers to the height of a location above sea level. In meteorology, elevation is an important factor that affects temperature.
- Elevation affects temperature in several ways:
    - Adiabatic cooling: The lower air pressure at higher elevations causes the air to expand, leading to a decrease in temperature. This is known as adiabatic cooling, and it occurs as the air rises and expands.
    - Insolation: At higher elevations, there is less atmospheric absorption of incoming solar radiation, which results in more insolation (solar radiation) reaching the surface. This can lead to increased surface temperature at higher elevations.
    - Latent heat: Higher elevations can experience lower atmospheric pressure and temperature, which can lead to increased evaporation and the release of latent heat, which can affect temperature.
    - Distance from the Earth's surface: The distance from the Earth's surface increases with elevation, which can reduce the impact of heat transfer from the Earth's surface and affect temperature.
  
<br>
  
- Ft 101 : **elevation__elevation**
    - Elevation
  

## **MJO Features**
___
- MJO in meteorology refers to the Madden-Julian Oscillation (MJO), which is a large-scale atmospheric phenomenon that occurs over the tropical regions of the Indian and Pacific Oceans. It is a dominant mode of variability in the tropical atmosphere and is characterized by a westward-propagating disturbance of clouds, rainfall, winds, and pressure.
- It affects temperature in the following ways:
    - The MJO is characterized by a westward-propagating disturbance of clouds, rainfall, winds, and pressure, and its passage can bring increased precipitation, cloud cover, and lower temperatures to affected regions. On the other hand, in regions where the MJO clears out the clouds and increases sunshine, temperatures can rise.
    - The MJO can also interact with other weather patterns, such as the El Niño Southern Oscillation (ENSO), to influence regional and global temperatures. For example, the MJO can amplify the warming or cooling effects of an El Niño or La Niña event.
- MJO phase: 
    - Refers to the specific stage or phase of the Madden-Julian Oscillation (MJO). The MJO is a large-scale atmospheric phenomenon that occurs over the tropical regions of the Indian and Pacific Oceans and is characterized by a westward-propagating disturbance of clouds, rainfall, winds, and pressure.
    - Is a way of categorizing the MJO into distinct stages of its cycle, with each phase having distinct features in terms of precipitation, cloud cover, winds, and pressure patterns. The exact definition of the MJO phases can vary, but in general, there are 8 to 10 phases that describe the westward progression of the MJO from one region to the next.
- MJO amplitude: 
    - Refers to the strength or intensity of the Madden-Julian Oscillation (MJO). The MJO is a large-scale atmospheric phenomenon that occurs over the tropical regions of the Indian and Pacific Oceans and is characterized by a westward-propagating disturbance of clouds, rainfall, winds, and pressure.
    - The amplitude of the MJO refers to the magnitude of the variability in the MJO signal, and is a measure of the strength of the MJO. A higher amplitude indicates a stronger MJO event, with more pronounced and widespread effects on weather patterns, such as increased precipitation, cloud cover, and winds. A lower amplitude indicates a weaker MJO event, with less impact on the weather.

<br>

- Ft 142 : **mjo1d__phase**
    - mjo phase
- Ft 143 : **mjo1d__amplitude**
    - mjo amplitude
    
## **MEI Features**
___
- MEI (Multivariate ENSO Index) is a measure of the strength and phase of the El Niño Southern Oscillation (ENSO) in the Pacific Ocean. ENSO is a large-scale ocean-atmosphere interaction that can have a significant impact on temperature and weather patterns around the world.
- MEI affects temperature in several ways:
    - Temperature anomalies: During an El Niño event, warm water from the western Pacific is transported to the eastern Pacific, resulting in a warming of the surface and atmosphere. This can result in temperature anomalies in affected regions.
    - Atmospheric circulation: ENSO can affect atmospheric circulation patterns, which can result in changes in temperature in affected regions.
    - Precipitation patterns: ENSO can also affect precipitation patterns, which can result in changes in temperature due to changes in evaporation and latent heat release
- MEI_Rank:
    - Refers to the ranking of the Multivariate ENSO Index (MEI), which measures the strength and phase of the El Niño Southern Oscillation (ENSO) in the Pacific Ocean. The MEI_Rank is a numerical value that indicates the relative strength of an ENSO event, with higher values indicating stronger ENSO events.
    - The MEI_Rank is used to compare the relative strength of different ENSO events over time, and to assess the potential impact of ENSO on temperature and weather patterns around the world. 
    - For example, a high MEI_Rank value during an El Niño event would indicate a stronger event and a potentially greater impact on temperature and weather patterns compared to a weaker event with a lower MEI_Rank value.
- MEI_Nip:
    - The MEI Niño Index Phase in meteorology refers to the phase of the Multivariate ENSO Index (MEI), which is a measure of the strength and phase of the El Niño Southern Oscillation (ENSO) in the Pacific Ocean. The MEI Niño Index Phase is a numerical value that indicates whether the current phase of ENSO is El Niño (warm), La Niña (cold), or neutral.
    - ENSO is a naturally occurring ocean-atmosphere interaction that has a significant impact on global temperature and weather patterns. The MEI Niño Index Phase is used to assess the current state of ENSO and to predict its potential impact on temperature and weather patterns around the world.

<br>

- Ft 144 : **mei__mei**
    - Multivariate ENSO Index
- Ft 145 : **mei__meirank**
    - Multivariate ENSO Index Rank
- Ft 146 : **mei__nip**
    - Multivariate ENSO Index Niño Index Phase

### **Sst Features**
___
- Sea Surface Temperature (SST) refers to the temperature of the upper layer of the ocean, typically measured at the surface. In meteorology, SST is an important factor that affects temperature.
- SST affects temperature in several ways:
    - Heat exchange: The ocean acts as a large heat sink, absorbing and releasing heat through processes such as evaporation, convection, and radiation. Changes in SST can result in heat exchange between the ocean and atmosphere, which can affect temperature.
    - Ocean currents: Ocean currents can transport warm or cool water from one region to another, which can result in changes in temperature.
    - Latent heat: The ocean can also release latent heat through processes such as evaporation, which can affect temperature by releasing energy into the atmosphere.
    - Atmospheric moisture: The ocean can also affect atmospheric moisture levels through processes such as evaporation, which can result in changes in temperature through the greenhouse effect.

<br>

- Ft 157 to Ft 166: **sst-2010-1 to sst-2010-10**
    - sea surface temperature

### **Icec Features**
___
- Sea ice concentration refers to the amount of sea ice present in a particular region, usually expressed as a percentage of the total area covered by sea ice. In meteorology, sea ice concentration is an important factor that affects temperature.
- Sea ice concentration affects temperature in several ways:
    - Albedo effect: Sea ice reflects a significant portion of incoming solar radiation back into the atmosphere, which can result in cooling of the surface and atmosphere.
    - Heat exchange: The presence of sea ice can reduce heat exchange between the ocean and atmosphere, which can affect temperature.
    - Latent heat: The formation and melting of sea ice can release and absorb latent heat, which can affect temperature by releasing or absorbing energy into the atmosphere.
    - Insolation: The presence of sea ice can also affect insolation (incoming solar radiation) by reducing the amount of radiation that reaches the surface.

<br>

- Ft 177 : **icec-2010-1 to icec-2010-10**
    - sea ice concentration
    

## **The Target Feature**

- Target : contest-tmp2m-14d__tmp2m
    - "TMP2m" typically refers to the "2-meter temperature," which is a measurement of the air temperature at a height of 2 meters above the surface of the Earth. 
    - This is a standard measurement used in meteorology and climatology to describe the temperature conditions of the lower atmosphere. 
    - The temperature at 2 meters is a commonly used variable for both observational and modeling purposes
    - This is because it is a good indicator of near-surface temperature conditions that can impact human activities, such as agriculture and transportation.
    - In this competition the target refers to the to the arithmetic mean of the maximum and minimum temperature over the next 14 days, for each location and start date.
    - Makes this a continous regression problem.
    